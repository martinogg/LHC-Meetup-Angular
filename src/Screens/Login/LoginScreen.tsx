import React, { Component } from 'react'
import { Button, Platform, StyleSheet, Text, View } from 'react-native'
import { NavigationScreenProp } from 'react-navigation'

import { StackActions } from 'react-navigation'

const instructions = Platform.select({
    ios: 'Press Cmd+R to reload,\n' + 'Cmd+D or shake for dev menu',
    android:
        'Double tap R on your keyboard to reload,\n' +
        'Shake or press menu button for dev menu',
});

interface Props { navigation: NavigationScreenProp<any, any> }
export class LoginScreen extends Component<Props, object> {

    public render() {

        const { navigate } = this.props.navigation;

        return (
            <View style={styles.container}>
                <Text style={styles.welcome}>Welcome to Login Screen!</Text>
                <Text style={styles.instructions}>To get started, edit App.tsx</Text>
                <Text style={styles.instructions}>{instructions}</Text>
                <Button title="Register"
                    onPress={() => navigate('Register')}
                />
                <Button title="Home Screen"
                    onPress={() => this.goToHomeScreen()} 
                />
                <Button title="EditDetails Screen"
                    onPress={() => this.pushEditDetailsScreen()} 
                />
            </View>
        );
    }

    private pushEditDetailsScreen() {
        this.props.navigation.navigate('EditDetails')
    }

    private goToHomeScreen() {
        
        this.props.navigation.replace('Home')

    }
}
// onPress={() => navigate('Register', { name: 'Jane' })} -- TODO: To Add params between screens
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    welcome: {
        fontSize: 20,
        textAlign: 'center',
        margin: 10,
    },
    instructions: {
        textAlign: 'center',
        color: '#333333',
        marginBottom: 5,
    },
});
