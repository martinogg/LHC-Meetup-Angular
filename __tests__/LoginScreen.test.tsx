import { shallow } from 'enzyme';
import React from 'react';
import 'react-native';
import { Button } from 'react-native'

import { LoginScreen } from '../src/Screens/Login/LoginScreen';

import renderer from 'react-test-renderer';

it('should display LoginScreen with no errors', () => {

  const navigation = { navigate: jest.fn() };
  expect(renderer.create(<LoginScreen navigation={navigation} />)).toMatchSnapshot();
});

test('test onPress Login functionality', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<LoginScreen navigation={navigation} />);

  wrapper.find(Button).first().simulate('press')

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('Register')
});

test('test pushEditDetails()', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<LoginScreen navigation={navigation} />);
  const sut: any = wrapper.instance()

  sut.pushEditDetailsScreen = jest.fn()

  wrapper.find(Button).at(2).simulate('press')

  expect(sut.pushEditDetailsScreen).toHaveBeenCalledTimes(1)
});

test('test pushHome()', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<LoginScreen navigation={navigation} />);
  const sut: any = wrapper.instance()

  sut.pushHomeScreen = jest.fn()

  wrapper.find(Button).at(1).simulate('press')

  expect(sut.pushHomeScreen).toHaveBeenCalledTimes(1)
});

test('test pushHomeScreen function', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<LoginScreen navigation={navigation} />);
  const sut: any = wrapper.instance()
  sut.pushHomeScreen()

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('Home')
});

test('test pushEditDetailsScreen function', () => {

  const navigateFunc = jest.fn()
  const navigation = { navigate: navigateFunc };

  const wrapper = shallow(<LoginScreen navigation={navigation} />);
  const sut: any = wrapper.instance()
  sut.pushEditDetailsScreen()

  expect(navigateFunc).toHaveBeenCalledTimes(1)
  expect(navigateFunc).toHaveBeenCalledWith('EditDetails')
});