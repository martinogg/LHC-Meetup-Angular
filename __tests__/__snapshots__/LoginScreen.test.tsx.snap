// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display LoginScreen with no errors 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#2c3e50",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#2c3e50",
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontWeight": "700",
              "textAlign": "center",
            },
            Object {
              "fontSize": 20,
            },
          ]
        }
      >
        Lets Have Coffee aims to co-ordinate people so they can meet one-on-one in real life for informal discussions about their shared interests
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          placeholderTextColor="rgba(225,225,225,0.7)"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "backgroundColor": "rgba(225,225,225,0.2)",
              "color": "#fff",
              "height": 40,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          placeholderTextColor="rgba(225,225,225,0.7)"
          rejectResponderTermination={true}
          returnKeyType="go"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(225,225,225,0.2)",
              "color": "#fff",
              "height": 40,
              "marginBottom": 10,
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2980b6",
              "opacity": 1,
              "paddingVertical": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2980b6",
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        CREATE NEW ACCOUNT
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
